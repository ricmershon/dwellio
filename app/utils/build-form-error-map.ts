import { ZodError } from "zod";

export const buildFormErrorMap = (issues: ZodError['issues']) => {
    console.log(issues);
    const errorMap: Record<string, string[]> = {};

    for (const issue of issues) {
        const path = issue.path.join('.');
        if (!errorMap[path]) {
            errorMap[path] = [];
        }
        errorMap[path].push(issue.message);
    }

    return structureErrors(errorMap);
};

type ErrorMap = Record<string, string[]>;

interface StructuredErrorMap {
    location?: Record<string, string[]>;
    rates?: Record<string, string[]>;
    sellerInfo?: Record<string, string[]>;
    [key: string]: Record<string, string[]> | string[] | undefined;
}

function structureErrors(flatErrors: ErrorMap): StructuredErrorMap {
    const structured: StructuredErrorMap = {};

    for (const [key, messages] of Object.entries(flatErrors)) {
        if (key.startsWith("location.")) {
            structured.location = structured.location || {};
            const field = key.split(".")[1];
            structured.location[field] = messages;
        } else if (key.startsWith("rates.")) {
            structured.rates = structured.rates || {};
            const field = key.split(".")[1];
            structured.rates[field] = messages;
        } else if (key.startsWith("sellerInfo.")) {
            structured.sellerInfo = structured.sellerInfo || {};
            const field = key.split(".")[1];
            structured.sellerInfo[field] = messages;
        } else {
            structured[key] = messages;
        }
    }

    return structured;
};