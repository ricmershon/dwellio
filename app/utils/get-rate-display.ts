import { Rates } from "@/app/types/definitions";

export const getRateDisplay = (rates: Rates) => {
    if (rates.monthly) {
        return `$${rates.monthly.toLocaleString()}/mo`;
    } else if (rates.weekly) {
        return `$${rates.weekly.toLocaleString()}/wk`;
    } else if (rates.nightly) {
        return `$${rates.nightly.toLocaleString()}/night`;
    }
};