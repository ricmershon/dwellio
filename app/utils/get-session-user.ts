import { getServerSession } from "next-auth/next";
import { authConfig } from "@/app/config/auth-config";

export const getSessionUser = async () => {
    const session = await getServerSession(authConfig);
    if (!session || !session.user) {
        return null;
    }
    return session.user
}