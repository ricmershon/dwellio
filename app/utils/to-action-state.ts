import { ActionState } from "@/app/lib/definitions";

export const toActionState = (
    message: string,
    status: "SUCCESS" | "ERROR",
    isBookmarked?: boolean
): ActionState => ({ message, status, isBookmarked });
