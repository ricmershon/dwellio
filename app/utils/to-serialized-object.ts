import { MessageDocument, PropertyDocument } from "@/app/models";

// TODO: need a cleaner solution
export const toSerializedOjbect = (object: PropertyDocument | PropertyDocument[] | MessageDocument) => (
    JSON.parse(JSON.stringify(object))
);