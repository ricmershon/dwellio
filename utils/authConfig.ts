import Google from "next-auth/providers/google";

export const authConfig = {
    providers: [
        Google({
            clientId: process.env.GOOGLE_CLIENT_ID!,
            clientSecret: process.env.GOOGLE_CLIENT_SECRET!,
            authorization: {
                params: {
                    prompt: 'consent',
                    access_type: 'offline',
                    response_type: 'code'
                }
            }
        })
    ],
    callbacks: {
        // Invoked on successful sign in
        async signIn({ profile }) {
            // 1. Connect to database
            // 2. Check if user exists
            // 3. If not then create user
            // 4. return true to allow sign in
        },
        // Session callback function that modifies the session object
        async session({ session }) {
            // 1. Get user from database
            // 2. Assign user id from the session
            // 3. Return session
        }
    }
};