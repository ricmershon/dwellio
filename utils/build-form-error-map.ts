import { ZodError } from "zod";
import { FormErrorsType } from "@/types/types";

// TODO: document functions in this file.
export const buildFormErrorMap = (issues: ZodError['issues']) => {
    const errorMap: Record<string, string[]> = {};

    for (const issue of issues) {
        const path = issue.path.join('.');
        if (!errorMap[path]) {
            errorMap[path] = [];
        }
        errorMap[path].push(issue.message);
    }

    return structureErrors(errorMap);
};

type ErrorMap = Record<string, string[]>;

export interface StructuredFormErrorMap {
    location?: Record<string, string[]>;
    rates?: Record<string, string[]>;
    sellerInfo?: Record<string, string[]>;
    [key: string]: FormErrorsType | undefined;
}

function structureErrors(flatErrors: ErrorMap): StructuredFormErrorMap {
    const structured: StructuredFormErrorMap = {};

    for (const [key, messages] of Object.entries(flatErrors)) {
        if (key.startsWith("location.")) {
            structured.location = structured.location || {};
            const field = key.split(".")[1];
            structured.location[field] = messages;
        } else if (key.startsWith("rates.")) {
            structured.rates = structured.rates || {};
            const field = key.split(".")[1];
            structured.rates[field] = messages;
        } else if (key.startsWith("sellerInfo.")) {
            structured.sellerInfo = structured.sellerInfo || {};
            const field = key.split(".")[1];
            structured.sellerInfo[field] = messages;
        } else {
            structured[key] = messages;
        }
    }

    return structured;
};