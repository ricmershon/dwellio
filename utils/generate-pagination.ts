/**
 * Generates pagination array with ellipsis for efficient navigation
 * @param currentPage - Current active page (1-based)
 * @param totalPages - Total number of pages available
 * @returns Array of page numbers and ellipsis strings
 */
export const generatePagination = (currentPage: number, totalPages: number) => {
    // Input validation
    if (typeof currentPage !== 'number' || typeof totalPages !== 'number') {
        console.warn('generatePagination: currentPage and totalPages must be numbers');
        return [1];
    }
    
    if (!Number.isInteger(currentPage) || !Number.isInteger(totalPages)) {
        console.warn('generatePagination: currentPage and totalPages must be integers');
        return [1];
    }
    
    if (totalPages < 1) {
        console.warn('generatePagination: totalPages must be at least 1');
        return [1];
    }
    
    if (currentPage < 1) {
        console.warn('generatePagination: currentPage must be at least 1, defaulting to page 1');
        currentPage = 1;
    }
    
    if (currentPage > totalPages) {
        console.warn(`generatePagination: currentPage (${currentPage}) exceeds totalPages (${totalPages}), defaulting to last page`);
        currentPage = totalPages;
    }

    try {
        /**
         * Return pagination without ellipsis if total number of pages is 5 or less.
         */
        if (totalPages <= 5) {
            return Array.from({ length: totalPages }, (_, i) => i + 1);
        }

        /**
         * Return "1", "2", "3", an ellipsis and the last 2 pages if the current page
         * is one of the first 3.
         */
        if (currentPage <= 3) {
            return [1, 2, 3, "...", totalPages - 1, totalPages];
        }

        /**
         * Return "1", "2", an ellipsis and the last 3 pages if the current page is
         * one of the last 3 pages.
         */
        if (currentPage >= totalPages - 2) {
            return [1, 2, "...", totalPages - 2, totalPages - 1, totalPages];
        }

        /**
         * Return "1", an ellipsis, the current pages and its neighbors, another
         * ellipsis and the last page if the current page is somewhere in the
         * middle.
         */
        return [
            1,
            "...",
            currentPage - 1,
            currentPage,
            currentPage + 1,
            "...",
            totalPages,
        ];
    } catch (error) {
        console.error('generatePagination: error generating pagination:', error);
        // Return safe fallback pagination
        return [1];
    }
};