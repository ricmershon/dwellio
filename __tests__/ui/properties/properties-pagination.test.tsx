import React from 'react';
import { render, screen, createNextNavigationMock } from '@/__tests__/test-utils';
import PropertiesPagination from '@/ui/properties/properties-pagination';
import { generatePagination } from '@/utils/generate-pagination';

// Mock Next.js navigation hooks
jest.mock('next/navigation', () => ({
    ...createNextNavigationMock(),
    usePathname: jest.fn(() => '/properties'),
}));

// Mock clsx to return joined class names
jest.mock('clsx', () => ((...args: unknown[]) => 
    args.filter(Boolean).join(' ')
));

// Mock Link component
jest.mock('next/link', () => ({
    __esModule: true,
    default: ({ children, href, className }: { children: React.ReactNode; href: string; className?: string }) => (
        <a href={href} className={className} data-testid="pagination-link">
            {children}
        </a>
    ),
}));

// Mock PaginationArrow subcomponent
jest.mock('@/ui/shared/pagination-arrow', () => ({
    __esModule: true,
    default: ({ direction, href, isDisabled }: { direction: string; href: string; isDisabled?: boolean }) => (
        <div 
            data-testid={`pagination-arrow-${direction}`}
            data-href={href}
            data-disabled={isDisabled}
            aria-disabled={isDisabled}
        >
            {direction === 'left' ? '←' : '→'}
        </div>
    ),
}));

// Mock PaginationNumber subcomponent
jest.mock('@/ui/shared/pagination-number', () => ({
    __esModule: true,
    default: ({ page, href, isActive, position }: { 
        page: number | string; 
        href: string; 
        isActive: boolean; 
        position?: string;
    }) => (
        <div 
            data-testid="pagination-number"
            data-page={page}
            data-href={href}
            data-active={isActive}
            data-position={position}
        >
            {page}
        </div>
    ),
}));

// Mock generatePagination utility - we'll test this separately
jest.mock('@/utils/generate-pagination');
const mockGeneratePagination = generatePagination as jest.MockedFunction<typeof generatePagination>;

// Get mocks from the navigation mock
const { mockPush, mockReplace, mockSearchParams } = jest.requireMock('next/navigation');

describe('PropertiesPagination', () => {
    const defaultProps = {
        currentPage: 1,
        totalPages: 10,
    };

    beforeEach(() => {
        jest.clearAllMocks();
        // Clear URLSearchParams manually (it doesn't have a clear() method)
        Array.from(mockSearchParams.keys()).forEach(key => {
            mockSearchParams.delete(key);
        });
        // Reset pathname mock to default
        const { usePathname } = jest.mocked(jest.requireMock('next/navigation'));
        usePathname.mockReturnValue('/properties');
        // Default mock implementation
        mockGeneratePagination.mockReturnValue([1, 2, 3, '...', 9, 10]);
    });

    describe('Component Structure', () => {
        it('should render main container with correct class', () => {
            const { container } = render(<PropertiesPagination {...defaultProps} />);
            
            const mainContainer = container.querySelector('.inline-flex');
            expect(mainContainer).toBeInTheDocument();
            expect(mainContainer).toHaveClass('inline-flex');
        });

        it('should render left arrow, pagination numbers, and right arrow', () => {
            render(<PropertiesPagination {...defaultProps} />);
            
            expect(screen.getByTestId('pagination-arrow-left')).toBeInTheDocument();
            expect(screen.getByTestId('pagination-arrow-right')).toBeInTheDocument();
            expect(screen.getAllByTestId('pagination-number')).toHaveLength(6); // [1, 2, 3, '...', 9, 10]
        });

        it('should render pagination numbers container with correct classes', () => {
            const { container } = render(<PropertiesPagination {...defaultProps} />);
            
            const numbersContainer = container.querySelector('.flex.-space-x-px');
            expect(numbersContainer).toBeInTheDocument();
        });
    });

    describe('URL Generation', () => {
        it('should create URLs with page parameter', () => {
            render(<PropertiesPagination currentPage={2} totalPages={5} />);
            
            const leftArrow = screen.getByTestId('pagination-arrow-left');
            const rightArrow = screen.getByTestId('pagination-arrow-right');
            
            expect(leftArrow).toHaveAttribute('data-href', '/properties?page=1');
            expect(rightArrow).toHaveAttribute('data-href', '/properties?page=3');
        });

        it('should preserve existing search parameters', () => {
            mockSearchParams.set('search', 'villa');
            mockSearchParams.set('location', 'miami');
            
            render(<PropertiesPagination currentPage={3} totalPages={8} />);
            
            const leftArrow = screen.getByTestId('pagination-arrow-left');
            const rightArrow = screen.getByTestId('pagination-arrow-right');
            
            expect(leftArrow.getAttribute('data-href')).toContain('search=villa');
            expect(leftArrow.getAttribute('data-href')).toContain('location=miami');
            expect(leftArrow.getAttribute('data-href')).toContain('page=2');
            
            expect(rightArrow.getAttribute('data-href')).toContain('search=villa');
            expect(rightArrow.getAttribute('data-href')).toContain('location=miami');
            expect(rightArrow.getAttribute('data-href')).toContain('page=4');
        });

        it('should handle different pathname', () => {
            const { usePathname } = jest.mocked(jest.requireMock('next/navigation'));
            usePathname.mockReturnValue('/properties/featured');
            
            render(<PropertiesPagination currentPage={1} totalPages={5} />);
            
            const rightArrow = screen.getByTestId('pagination-arrow-right');
            expect(rightArrow).toHaveAttribute('data-href', '/properties/featured?page=2');
        });
    });

    describe('Arrow States', () => {
        it('should disable left arrow on first page', () => {
            render(<PropertiesPagination currentPage={1} totalPages={5} />);
            
            const leftArrow = screen.getByTestId('pagination-arrow-left');
            expect(leftArrow).toHaveAttribute('data-disabled', 'true');
        });

        it('should disable right arrow on last page', () => {
            render(<PropertiesPagination currentPage={5} totalPages={5} />);
            
            const rightArrow = screen.getByTestId('pagination-arrow-right');
            expect(rightArrow).toHaveAttribute('data-disabled', 'true');
        });

        it('should enable both arrows on middle pages', () => {
            render(<PropertiesPagination currentPage={3} totalPages={5} />);
            
            const leftArrow = screen.getByTestId('pagination-arrow-left');
            const rightArrow = screen.getByTestId('pagination-arrow-right');
            
            expect(leftArrow).toHaveAttribute('data-disabled', 'false');
            expect(rightArrow).toHaveAttribute('data-disabled', 'false');
        });

        it('should handle single page correctly', () => {
            render(<PropertiesPagination currentPage={1} totalPages={1} />);
            
            const leftArrow = screen.getByTestId('pagination-arrow-left');
            const rightArrow = screen.getByTestId('pagination-arrow-right');
            
            expect(leftArrow).toHaveAttribute('data-disabled', 'true');
            expect(rightArrow).toHaveAttribute('data-disabled', 'true');
        });
    });

    describe('Pagination Number Generation', () => {
        it('should call generatePagination with correct parameters', () => {
            render(<PropertiesPagination currentPage={5} totalPages={12} />);
            
            expect(mockGeneratePagination).toHaveBeenCalledWith(5, 12);
        });

        it('should render pagination numbers with correct props', () => {
            mockGeneratePagination.mockReturnValue([1, '...', 4, 5, 6, '...', 10]);
            
            render(<PropertiesPagination currentPage={5} totalPages={10} />);
            
            const numbers = screen.getAllByTestId('pagination-number');
            expect(numbers).toHaveLength(7);
            
            // Check first number
            expect(numbers[0]).toHaveAttribute('data-page', '1');
            expect(numbers[0]).toHaveAttribute('data-active', 'false');
            expect(numbers[0]).toHaveAttribute('data-position', 'first');
            
            // Check ellipsis (middle position)
            expect(numbers[1]).toHaveAttribute('data-page', '...');
            expect(numbers[1]).toHaveAttribute('data-position', 'middle');
            
            // Check active page
            expect(numbers[3]).toHaveAttribute('data-page', '5');
            expect(numbers[3]).toHaveAttribute('data-active', 'true');
            
            // Check last number
            expect(numbers[6]).toHaveAttribute('data-page', '10');
            expect(numbers[6]).toHaveAttribute('data-position', 'last');
        });

        it('should handle single page pagination', () => {
            mockGeneratePagination.mockReturnValue([1]);
            
            render(<PropertiesPagination currentPage={1} totalPages={1} />);
            
            const numbers = screen.getAllByTestId('pagination-number');
            expect(numbers).toHaveLength(1);
            expect(numbers[0]).toHaveAttribute('data-position', 'single');
        });

        it('should generate unique keys for pagination numbers', () => {
            mockGeneratePagination.mockReturnValue([1, '...', 3, '...', 5]);
            
            const { container } = render(<PropertiesPagination currentPage={3} totalPages={5} />);
            
            // Each pagination number should have a unique key (page-index format)
            const numbers = container.querySelectorAll('[data-testid="pagination-number"]');
            expect(numbers).toHaveLength(5);
        });
    });

    describe('Position Logic', () => {
        it('should set correct positions for multiple pages', () => {
            mockGeneratePagination.mockReturnValue([1, 2, 3, 4, 5]);
            
            render(<PropertiesPagination currentPage={3} totalPages={5} />);
            
            const numbers = screen.getAllByTestId('pagination-number');
            
            expect(numbers[0]).toHaveAttribute('data-position', 'first');
            expect(numbers[1]).not.toHaveAttribute('data-position'); // middle pages have no position
            expect(numbers[2]).not.toHaveAttribute('data-position');
            expect(numbers[3]).not.toHaveAttribute('data-position');
            expect(numbers[4]).toHaveAttribute('data-position', 'last');
        });

        it('should handle ellipsis positions correctly', () => {
            mockGeneratePagination.mockReturnValue([1, '...', 5, 6, 7, '...', 12]);
            
            render(<PropertiesPagination currentPage={6} totalPages={12} />);
            
            const numbers = screen.getAllByTestId('pagination-number');
            
            expect(numbers[0]).toHaveAttribute('data-position', 'first'); // 1
            expect(numbers[1]).toHaveAttribute('data-position', 'middle'); // '...'
            expect(numbers[5]).toHaveAttribute('data-position', 'middle'); // '...'
            expect(numbers[6]).toHaveAttribute('data-position', 'last'); // 12
        });
    });

    describe('Props Validation', () => {
        it('should handle zero totalPages', () => {
            mockGeneratePagination.mockReturnValue([]);
            
            render(<PropertiesPagination currentPage={1} totalPages={0} />);
            
            expect(mockGeneratePagination).toHaveBeenCalledWith(1, 0);
            expect(screen.queryAllByTestId('pagination-number')).toHaveLength(0);
        });

        it('should handle negative currentPage', () => {
            mockGeneratePagination.mockReturnValue([1, 2, 3]);
            
            render(<PropertiesPagination currentPage={-1} totalPages={3} />);
            
            expect(mockGeneratePagination).toHaveBeenCalledWith(-1, 3);
        });

        it('should handle currentPage greater than totalPages', () => {
            mockGeneratePagination.mockReturnValue([1, 2, 3]);
            
            render(<PropertiesPagination currentPage={5} totalPages={3} />);
            
            const rightArrow = screen.getByTestId('pagination-arrow-right');
            expect(rightArrow).toHaveAttribute('data-disabled', 'true');
        });

        it('should handle large page numbers', () => {
            mockGeneratePagination.mockReturnValue([1, '...', 998, 999, 1000]);
            
            render(<PropertiesPagination currentPage={999} totalPages={1000} />);
            
            expect(mockGeneratePagination).toHaveBeenCalledWith(999, 1000);
            
            const numbers = screen.getAllByTestId('pagination-number');
            expect(numbers[4]).toHaveAttribute('data-page', '1000');
        });
    });

    describe('Memoization', () => {
        it('should memoize pagination generation', () => {
            const { rerender } = render(<PropertiesPagination currentPage={3} totalPages={10} />);
            
            expect(mockGeneratePagination).toHaveBeenCalledTimes(1);
            
            // Re-render with same props - should not call generatePagination again due to useMemo
            rerender(<PropertiesPagination currentPage={3} totalPages={10} />);
            
            expect(mockGeneratePagination).toHaveBeenCalledTimes(1);
        });

        it('should regenerate pagination when props change', () => {
            const { rerender } = render(<PropertiesPagination currentPage={3} totalPages={10} />);
            
            expect(mockGeneratePagination).toHaveBeenCalledWith(3, 10);
            
            rerender(<PropertiesPagination currentPage={4} totalPages={10} />);
            
            expect(mockGeneratePagination).toHaveBeenCalledWith(4, 10);
            expect(mockGeneratePagination).toHaveBeenCalledTimes(2);
        });

        it('should memoize createPageURL callback', () => {
            const { rerender } = render(<PropertiesPagination currentPage={2} totalPages={5} />);
            
            const initialLeftArrow = screen.getByTestId('pagination-arrow-left');
            const initialHref = initialLeftArrow.getAttribute('data-href');
            
            // Re-render with same search params - URL should be memoized
            rerender(<PropertiesPagination currentPage={2} totalPages={5} />);
            
            const rerenderedLeftArrow = screen.getByTestId('pagination-arrow-left');
            expect(rerenderedLeftArrow.getAttribute('data-href')).toBe(initialHref);
        });
    });

    describe('Integration with Dependencies', () => {
        it('should pass correct props to PaginationArrow components', () => {
            render(<PropertiesPagination currentPage={3} totalPages={8} />);
            
            const leftArrow = screen.getByTestId('pagination-arrow-left');
            const rightArrow = screen.getByTestId('pagination-arrow-right');
            
            expect(leftArrow).toHaveAttribute('data-href', '/properties?page=2');
            expect(leftArrow).toHaveAttribute('data-disabled', 'false');
            
            expect(rightArrow).toHaveAttribute('data-href', '/properties?page=4');
            expect(rightArrow).toHaveAttribute('data-disabled', 'false');
        });

        it('should pass correct props to PaginationNumber components', () => {
            mockGeneratePagination.mockReturnValue([1, 2, 3]);
            
            render(<PropertiesPagination currentPage={2} totalPages={3} />);
            
            const numbers = screen.getAllByTestId('pagination-number');
            
            expect(numbers[0]).toHaveAttribute('data-page', '1');
            expect(numbers[0]).toHaveAttribute('data-href', '/properties?page=1');
            expect(numbers[0]).toHaveAttribute('data-active', 'false');
            
            expect(numbers[1]).toHaveAttribute('data-page', '2');
            expect(numbers[1]).toHaveAttribute('data-href', '/properties?page=2');
            expect(numbers[1]).toHaveAttribute('data-active', 'true');
            
            expect(numbers[2]).toHaveAttribute('data-page', '3');
            expect(numbers[2]).toHaveAttribute('data-href', '/properties?page=3');
            expect(numbers[2]).toHaveAttribute('data-active', 'false');
        });
    });

    describe('Accessibility', () => {
        it('should have accessible structure', () => {
            render(<PropertiesPagination currentPage={2} totalPages={5} />);
            
            const leftArrow = screen.getByTestId('pagination-arrow-left');
            const rightArrow = screen.getByTestId('pagination-arrow-right');
            
            expect(leftArrow).toHaveAttribute('aria-disabled', 'false');
            expect(rightArrow).toHaveAttribute('aria-disabled', 'false');
        });

        it('should properly mark disabled arrows', () => {
            render(<PropertiesPagination currentPage={1} totalPages={3} />);
            
            const leftArrow = screen.getByTestId('pagination-arrow-left');
            expect(leftArrow).toHaveAttribute('aria-disabled', 'true');
        });
    });

    describe('Snapshots', () => {
        it('should match snapshot for first page', () => {
            mockGeneratePagination.mockReturnValue([1, 2, 3, '...', 10]);
            
            const { container } = render(<PropertiesPagination currentPage={1} totalPages={10} />);
            
            expect(container.firstChild).toMatchSnapshot();
        });

        it('should match snapshot for middle page', () => {
            mockGeneratePagination.mockReturnValue([1, '...', 4, 5, 6, '...', 10]);
            
            const { container } = render(<PropertiesPagination currentPage={5} totalPages={10} />);
            
            expect(container.firstChild).toMatchSnapshot();
        });

        it('should match snapshot for last page', () => {
            mockGeneratePagination.mockReturnValue([1, 2, '...', 8, 9, 10]);
            
            const { container } = render(<PropertiesPagination currentPage={10} totalPages={10} />);
            
            expect(container.firstChild).toMatchSnapshot();
        });

        it('should match snapshot for single page', () => {
            mockGeneratePagination.mockReturnValue([1]);
            
            const { container } = render(<PropertiesPagination currentPage={1} totalPages={1} />);
            
            expect(container.firstChild).toMatchSnapshot();
        });
    });
});