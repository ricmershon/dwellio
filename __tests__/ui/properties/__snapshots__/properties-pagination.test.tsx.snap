// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`PropertiesPagination Snapshots should match snapshot for first page 1`] = `
<div
  class="inline-flex"
>
  <div
    aria-disabled="true"
    data-disabled="true"
    data-href="/properties?page=0"
    data-testid="pagination-arrow-left"
  >
    ←
  </div>
  <div
    class="flex -space-x-px"
  >
    <div
      data-active="true"
      data-href="/properties?page=1"
      data-page="1"
      data-position="first"
      data-testid="pagination-number"
    >
      1
    </div>
    <div
      data-active="false"
      data-href="/properties?page=2"
      data-page="2"
      data-testid="pagination-number"
    >
      2
    </div>
    <div
      data-active="false"
      data-href="/properties?page=3"
      data-page="3"
      data-testid="pagination-number"
    >
      3
    </div>
    <div
      data-active="false"
      data-href="/properties?page=..."
      data-page="..."
      data-position="middle"
      data-testid="pagination-number"
    >
      ...
    </div>
    <div
      data-active="false"
      data-href="/properties?page=10"
      data-page="10"
      data-position="last"
      data-testid="pagination-number"
    >
      10
    </div>
  </div>
  <div
    aria-disabled="false"
    data-disabled="false"
    data-href="/properties?page=2"
    data-testid="pagination-arrow-right"
  >
    →
  </div>
</div>
`;

exports[`PropertiesPagination Snapshots should match snapshot for last page 1`] = `
<div
  class="inline-flex"
>
  <div
    aria-disabled="false"
    data-disabled="false"
    data-href="/properties?page=9"
    data-testid="pagination-arrow-left"
  >
    ←
  </div>
  <div
    class="flex -space-x-px"
  >
    <div
      data-active="false"
      data-href="/properties?page=1"
      data-page="1"
      data-position="first"
      data-testid="pagination-number"
    >
      1
    </div>
    <div
      data-active="false"
      data-href="/properties?page=2"
      data-page="2"
      data-testid="pagination-number"
    >
      2
    </div>
    <div
      data-active="false"
      data-href="/properties?page=..."
      data-page="..."
      data-position="middle"
      data-testid="pagination-number"
    >
      ...
    </div>
    <div
      data-active="false"
      data-href="/properties?page=8"
      data-page="8"
      data-testid="pagination-number"
    >
      8
    </div>
    <div
      data-active="false"
      data-href="/properties?page=9"
      data-page="9"
      data-testid="pagination-number"
    >
      9
    </div>
    <div
      data-active="true"
      data-href="/properties?page=10"
      data-page="10"
      data-position="last"
      data-testid="pagination-number"
    >
      10
    </div>
  </div>
  <div
    aria-disabled="true"
    data-disabled="true"
    data-href="/properties?page=11"
    data-testid="pagination-arrow-right"
  >
    →
  </div>
</div>
`;

exports[`PropertiesPagination Snapshots should match snapshot for middle page 1`] = `
<div
  class="inline-flex"
>
  <div
    aria-disabled="false"
    data-disabled="false"
    data-href="/properties?page=4"
    data-testid="pagination-arrow-left"
  >
    ←
  </div>
  <div
    class="flex -space-x-px"
  >
    <div
      data-active="false"
      data-href="/properties?page=1"
      data-page="1"
      data-position="first"
      data-testid="pagination-number"
    >
      1
    </div>
    <div
      data-active="false"
      data-href="/properties?page=..."
      data-page="..."
      data-position="middle"
      data-testid="pagination-number"
    >
      ...
    </div>
    <div
      data-active="false"
      data-href="/properties?page=4"
      data-page="4"
      data-testid="pagination-number"
    >
      4
    </div>
    <div
      data-active="true"
      data-href="/properties?page=5"
      data-page="5"
      data-testid="pagination-number"
    >
      5
    </div>
    <div
      data-active="false"
      data-href="/properties?page=6"
      data-page="6"
      data-testid="pagination-number"
    >
      6
    </div>
    <div
      data-active="false"
      data-href="/properties?page=..."
      data-page="..."
      data-position="middle"
      data-testid="pagination-number"
    >
      ...
    </div>
    <div
      data-active="false"
      data-href="/properties?page=10"
      data-page="10"
      data-position="last"
      data-testid="pagination-number"
    >
      10
    </div>
  </div>
  <div
    aria-disabled="false"
    data-disabled="false"
    data-href="/properties?page=6"
    data-testid="pagination-arrow-right"
  >
    →
  </div>
</div>
`;

exports[`PropertiesPagination Snapshots should match snapshot for single page 1`] = `
<div
  class="inline-flex"
>
  <div
    aria-disabled="true"
    data-disabled="true"
    data-href="/properties?page=0"
    data-testid="pagination-arrow-left"
  >
    ←
  </div>
  <div
    class="flex -space-x-px"
  >
    <div
      data-active="true"
      data-href="/properties?page=1"
      data-page="1"
      data-position="single"
      data-testid="pagination-number"
    >
      1
    </div>
  </div>
  <div
    aria-disabled="true"
    data-disabled="true"
    data-href="/properties?page=2"
    data-testid="pagination-arrow-right"
  >
    →
  </div>
</div>
`;
